### PR TITLE
Theme validation: add style.css metadata validations

### PR DESCRIPTION
**Pull request by madhusudhand**
#### Changes proposed in this Pull Request:

This adds validations to style.css metadata values (missing and correctness).
It is observed that few themes are missing some of the required values or having invalid values.
This script should help find them.

#### Testing instructions:

* Try setting invalid, or empty values for the `style.css` metadata
* run the script `npm run validate:theme <theme-name>`
* It should throw errors or warnings

#### Related issue(s):

This is an extension of #8039

### CODE REVIEW

File: theme-utils.mjs
Comment (ajlende) [inline theme-utils.mjs]: I expect methods starting with `is` to return a `boolean`, but this returns `undefined` if valid or an `Object` if invalid. There's a few options to change this, and I don't have a strong preference for any particular one.

1. Rename all of them to something like `getVersionProblems`.
2. Make it work like AJV validators. It returns a boolean from validate and sets `this.errors` on the function when it is run. The consuming code checks `validator.errors` to handle them accordingly.
3. Return a boolean and throw the error. The consuming code can catch errors and handle them accordingly.
4. Return just the boolean and move the problem return value to its own value. Maybe by adding something like `getProblemData: ( sameArgsAsValidate ) => problemData` to `styleCssMetadata` after to the `type`.

It would be nice to extract the whole theme validator into its own package at some point and allow consumers to register their own validators, so it can be used in things like WordPress/community-themes. Maybe choose whatever you think would make the most sense in that case to make refactoring a bit easier later. Ultimately it doesn't matter that much now since we can make a final decision when we get there. 🤷

File: theme-utils.mjs
Comment (ajlende) [inline theme-utils.mjs]: I don't think I would add this to the main list of validators here.

I should have probably named `isSupportedWpVersion` something like `hasOfficialThemeJsonSchema` instead and had a better warning message to be more clear about that.

`5.9` is just the version where the official theme.json JSON schema was introduced.

Previously the goal was only validating theme.json files, so that's why the warning existed at the top level. Now it probably makes more sense as a warning for block themes only.

We can still validate that all Automattic/themes are block themes with a theme.json, but I'd rather do that with a `fs.existsSync( 'theme.json' )` check instead.

**Review by ajlende — COMMENTED** [review]
I like how this is organized. Just a couple minor comments that need addressing.

File: theme-utils.mjs
Comment (madhusudhand) [reply theme-utils.mjs]: renamed all the functions to start with `validate` and they are expected to return an object `{ isValid: boolean, problems: [object]`

> It would be nice to extract the whole theme validator into its own package at some point .... we can make a final decision when we get there. 🤷

That's a good idea. we can further refine it and extract all of it as a separate module which allows validators to be registered based on the repository need. 👍

File: theme-utils.mjs
Comment (madhusudhand) [reply theme-utils.mjs]: Makes sense. Reverted to previous validation with additional check of `theme.json` existence.

File: theme-utils.mjs
Comment (ajlende) [inline theme-utils.mjs]:
```suggestion
		const hasThemeJsonSupport = wpVersion && semver.valid( `${ wpVersion }.0` ) && semver.gte( `${ wpVersion }.0`, '5.9.0' )
```

File: theme-utils.mjs
Comment (ajlende) [inline theme-utils.mjs]: It seems the `/themes/{ slug }` URLs redirect to `/theme/{ slug }`, so maybe this should allow both or just `/theme`?

**Review by ajlende — APPROVED** [review]
Looks good to me. Just a couple minor things to tweak before merging.

I ran it for all themes—filtering out just the messages—to get a high-level overview of the existing problems in the repo.

```sh-session
$ node theme-utils.mjs validate-theme --format=json $(find . -name 'style.css' | awk -F/ '{print $2}' | uniq | sort | paste -s -d, -) | jq '.[].data[].message' | sort | uniq -c | sort -bgr
5987 "must NOT have additional properties"
 734 "must be object"
 604 "must be string"
 348 "the $schema version does not match style.css 'Requires at least' version"
 191 "must match exactly one schema in oneOf"
  84 "must be equal to constant"
  75 "must be equal to one of the allowed values"
  74 "the 'Requires at least' version does not support theme.json"
  70 "https://automattic.com is not a valid Automattic author URI"
  58 "missing Requires PHP header metadata"
  57 null
  53 "missing Tested up to header metadata"
  47 "Missing $schema URI: undefined"
  32 "missing Requires at least header metadata"
  29 "LICENSE is not a valid URI"
  25 "WordPress 4.9.6 is not a valid version"
  18 "the WordPress team is not a valid author"
  12 "missing Theme URI header metadata"
  12 "https://wordpress.org/themes/ is not a valid Automattic author URI"
  11 "https://wordpress.com/themes/ is not a valid Automattic author URI"
  10 "1.0 is not a valid version"
   9 "https://wordpress.org/ is not a valid Automattic author URI"
   8 "http://wordpress.com/themes/ is not a valid Automattic author URI"
   6 "provide a valid version value"
   6 "property name must be valid"
   5 "must be number"
   5 "1.1 is not a valid version"
   4 "1.2 is not a valid version"
   3 "5.6.2 is not a valid version"
   2 "must be >= 1"
   2 "https://wordpress.org is not a valid Automattic author URI"
   2 "http://wordpress.com/themes is not a valid Automattic author URI"
   2 "http://automattic.com is not a valid Automattic author URI"
   2 "4.9.6 is not a valid version"
   2 "1.3 is not a valid version"
   1 "the theme is missing a style.css file"
   1 "the Automattic is not a valid author"
   1 "must have required property 'version'"
   1 "must have required property 'slug'"
   1 "must have required property 'name'"
   1 "license.txt is not a valid URI"
   1 "https://www.wordpress.com is not a valid Automattic author URI"
   1 "https://wordpress.org/themes/rebalance/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.org/themes/curriculum is not a valid Automattic author URI"
   1 "https://wordpress.org/themes/alter is not a valid Automattic author URI"
   1 "https://wordpress.com/theme/zoologist is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/xanadu is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/winkel is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/vivre is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/varia is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/upsidedown/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/ueno is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/twentytwentytwo-swiss is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/twentytwentytwo-red is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/twentytwentytwo-pink is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/twentytwentytwo-mint is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/twentytwentytwo-blue is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/tu/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/tronar/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/trellick/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/tomoni/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/tenaz/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/stratford is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/stow is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/storia/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/stewart is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/startfit/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/spiel is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/spearhead-blocks/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/spearhead is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/sophisticated-business is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/skatepark is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/shhh/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/shawburn is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/seedlet-blocks is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/seedlet is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/russell is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/ron is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/rockfield is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/rivington is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/reverie/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/remote is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/redhill is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/rainfall/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/quadrat is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/professional-business is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/pomme/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/poesis/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/pixl/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/pieria is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/perenne is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/pendant is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/paimio/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/overlaid is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/optimismo/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/nook is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/negai/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/mysa/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/muscat/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/morden is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/montagna/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/modern-business is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/meraki is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/maywood is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/mayland-blocks is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/mayland is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/masu/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/marl is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/luxus is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/luminance is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/luce is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/lowfi/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/loudness/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/lois/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/loic is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/livro is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/lineup/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/leven is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/lativ/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/kiosko/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/kingsley is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/kigen is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/kentwood/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/kaze/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/kansei/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/jinjang/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/jackson is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/infield is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/indice is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/ici/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/hey/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/hever is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/heiwa is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/hari is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/happening/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/grammerone/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/geologist is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/friendly-business is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/freddie/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/fotograma/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/fewer is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/feelingood/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/farrow is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/exmoor/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/exford is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/eventual/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/epi/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/entry is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/elegant-business is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/dorna is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/disco/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/dalston is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/ctlg is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/creatio-2 is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/creatio is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/craftfully is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/covr is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/coutoire is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/cortado is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/common is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/chanson/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/calvin is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/calm-business is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/cakely/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/bute/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/bsoj is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/brompton is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/boxedbio is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/blockbase is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/blank-canvas-blocks is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/blank-canvas-3/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/blank-canvas is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/bitacora is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/bibliophile/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/bibimbap/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/bennett is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/beep is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/bedrock is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/barnsbury is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/barnett is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/bark is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/balasana is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/azur is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/awburn is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/attar is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/assembler is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/artly is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/archivo/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/archeo is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/arbutus is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/appleton is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/antonia is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/ames is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/alves is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/aldente is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/theme/adventurer/ is not a valid WordPress.com theme URI"
   1 "https://wordpress.com/ is not a valid Automattic author URI"
   1 "https://wordpress.com is not a valid Automattic author URI"
   1 "https://themeshaper.com/vows/ is not a valid WordPress.com theme URI"
   1 "https://themeshaper.com/vitrum/ is not a valid WordPress.com theme URI"
   1 "https://themeshaper.com/trailblazer is not a valid WordPress.com theme URI"
   1 "https://themeshaper.com/the-shore/ is not a valid WordPress.com theme URI"
   1 "https://themeshaper.com/surrealist/ is not a valid WordPress.com theme URI"
   1 "https://themeshaper.com/stageplay/ is not a valid WordPress.com theme URI"
   1 "https://themeshaper.com/specials/ is not a valid WordPress.com theme URI"
   1 "https://themeshaper.com/recipebook/ is not a valid WordPress.com theme URI"
   1 "https://themeshaper.com/portia/ is not a valid WordPress.com theme URI"
   1 "https://themeshaper.com/pierian/ is not a valid WordPress.com theme URI"
   1 "https://themeshaper.com/partygurl/ is not a valid WordPress.com theme URI"
   1 "https://themeshaper.com/outnow/ is not a valid WordPress.com theme URI"
   1 "https://themeshaper.com/otis/ is not a valid WordPress.com theme URI"
   1 "https://themeshaper.com/onyxpulse/ is not a valid WordPress.com theme URI"
   1 "https://themeshaper.com/olympique/ is not a valid WordPress.com theme URI"
   1 "https://themeshaper.com/mymenu/ is not a valid WordPress.com theme URI"
   1 "https://themeshaper.com/musee/ is not a valid WordPress.com theme URI"
   1 "https://themeshaper.com/moviescript/ is not a valid WordPress.com theme URI"
   1 "https://themeshaper.com/maverick/ is not a valid WordPress.com theme URI"
   1 "https://themeshaper.com/matrioska/ is not a valid WordPress.com theme URI"
   1 "https://themeshaper.com/leancv/ is not a valid WordPress.com theme URI"
   1 "https://themeshaper.com/inversum/ is not a valid WordPress.com theme URI"
   1 "https://themeshaper.com/hola/ is not a valid WordPress.com theme URI"
   1 "https://themeshaper.com/grammer/ is not a valid WordPress.com theme URI"
   1 "https://themeshaper.com/fontaine/ is not a valid WordPress.com theme URI"
   1 "https://themeshaper.com/fixmate/ is not a valid WordPress.com theme URI"
   1 "https://themeshaper.com/expo/ is not a valid WordPress.com theme URI"
   1 "https://themeshaper.com/erma/ is not a valid WordPress.com theme URI"
   1 "https://themeshaper.com/dark-academia/ is not a valid WordPress.com theme URI"
   1 "https://themeshaper.com/cookbook/ is not a valid WordPress.com theme URI"
   1 "https://themeshaper.com/brightblog/ is not a valid WordPress.com theme URI"
   1 "https://themeshaper.com/blissed/ is not a valid WordPress.com theme URI"
   1 "https://themeshaper.com/atrium/ is not a valid WordPress.com theme URI"
   1 "https://github.com/wordpress/the-menu/ is not a valid WordPress.com theme URI"
   1 "https://github.com/wordpress/stage/ is not a valid WordPress.com theme URI"
   1 "https://github.com/wordpress/raw/ is not a valid WordPress.com theme URI"
   1 "https://github.com/wordpress/programme/ is not a valid WordPress.com theme URI"
   1 "https://github.com/wordpress/organizer/ is not a valid WordPress.com theme URI"
   1 "https://github.com/wordpress/messagerie/ is not a valid WordPress.com theme URI"
   1 "https://github.com/wordpress/jazzers/ is not a valid WordPress.com theme URI"
   1 "https://github.com/wordpress/issue/ is not a valid WordPress.com theme URI"
   1 "https://github.com/wordpress/hall/ is not a valid WordPress.com theme URI"
   1 "https://github.com/wordpress/curriculum/ is not a valid WordPress.com theme URI"
   1 "https://github.com/wordpress/blogorama/ is not a valid WordPress.com theme URI"
   1 "https://github.com/wordpress/alter/ is not a valid WordPress.com theme URI"
   1 "https://github.com/wordpress/allez/ is not a valid WordPress.com theme URI"
   1 "https://github.com/Automattic/themes/tree/trunk/texty is not a valid WordPress.com theme URI"
   1 "https://github.com/Automattic/themes/tree/trunk/sunderland is not a valid WordPress.com theme URI"
   1 "https://github.com/Automattic/themes/tree/trunk/screenplay is not a valid WordPress.com theme URI"
   1 "https://github.com/Automattic/themes/tree/trunk/saison is not a valid WordPress.com theme URI"
   1 "https://github.com/Automattic/themes/tree/trunk/poema is not a valid WordPress.com theme URI"
   1 "https://github.com/Automattic/themes/tree/trunk/nested is not a valid WordPress.com theme URI"
   1 "https://github.com/Automattic/themes/tree/trunk/mpho is not a valid WordPress.com theme URI"
   1 "https://github.com/Automattic/themes/tree/trunk/jaida is not a valid WordPress.com theme URI"
   1 "https://github.com/Automattic/themes/tree/trunk/foam is not a valid WordPress.com theme URI"
   1 "https://github.com/Automattic/themes/tree/trunk/farmhouse is not a valid WordPress.com theme URI"
   1 "https://github.com/Automattic/themes/tree/trunk/dos is not a valid WordPress.com theme URI"
   1 "https://github.com/Automattic/themes/tree/trunk/didone is not a valid WordPress.com theme URI"
   1 "https://github.com/Automattic/themes/tree/trunk/block-canvas is not a valid WordPress.com theme URI"
   1 "https://github.com/Automattic/themes/tree/trunk/barnsbury23 is not a valid WordPress.com theme URI"
   1 "https://github.com/Automattic/themes/ is not a valid WordPress.com theme URI"
   1 "https://automatic.com is not a valid Automattic author URI"
   1 "http://www.andersnoren.se is not a valid Automattic author URI"
   1 "http://wordpress.com/themes/toujours/ is not a valid WordPress.com theme URI"
   1 "http://wordpress.com/themes/radcliffe-2 is not a valid WordPress.com theme URI"
   1 "http://wordpress.com/themes/libretto is not a valid WordPress.com theme URI"
   1 "http://wordpress.com/themes/libre-2/ is not a valid WordPress.com theme URI"
   1 "http://wordpress.com/themes/karuna/ is not a valid WordPress.com theme URI"
   1 "http://wordpress.com/themes/ixion/ is not a valid WordPress.com theme URI"
   1 "http://wordpress.com/themes/independent-publisher-2 is not a valid WordPress.com theme URI"
   1 "http://wordpress.com/themes/dyad-2/ is not a valid WordPress.com theme URI"
   1 "http://wordpress.com/themes/dara/ is not a valid WordPress.com theme URI"
   1 "http://wordpress.com/themes/button-2/ is not a valid WordPress.com theme URI"
   1 "http://wordpress.com/themes/affinity/ is not a valid WordPress.com theme URI"
   1 "http://wordpress.com/theme/textbook/ is not a valid WordPress.com theme URI"
   1 "http://wordpress.com/theme/lettre/ is not a valid WordPress.com theme URI"
   1 "http://theme.wordpress.com/themes/scratchpad is not a valid WordPress.com theme URI"
   1 "http://theme.wordpress.com/themes/lodestar is not a valid WordPress.com theme URI"
   1 "http://raamdev.com/ is not a valid Automattic author URI"
   1 "http://automattic.com/ is not a valid Automattic author URI"
   1 "Raam Dev is not a valid author"
   1 "Automattic, Inc is not a valid author"
   1 "Anders Norén is not a valid author"
   1 "6.6-RC4 is not a valid version"
   1 "5.7.2 is not a valid version"
   1 "1.4 is not a valid version"
   1 "1.17 is not a valid version"
   1 "/https://themeshaper.com/goodskin/ is not a valid WordPress.com theme URI"
   1 "/https://themeshaper.com/goodskin/ is not a valid URI"
```

File: theme-utils.mjs
Comment (madhusudhand) [reply theme-utils.mjs]: Good catch. I think we should allow both. fixed in [75fd4e9](https://github.com/Automattic/themes/pull/8068/commits/75fd4e9f2b2a9ad3802814050e10b41ce372a783)

File: theme-utils.mjs
Comment (madhusudhand) [reply theme-utils.mjs]: fixed in [75fd4e9](https://github.com/Automattic/themes/pull/8068/commits/75fd4e9f2b2a9ad3802814050e10b41ce372a783)

**Comment by vcanales** [comment]
@madhusudhand This seems to have introduced a bug in the workflow:

https://github.com/Automattic/themes/actions/runs/10861752016/job/30144004760?pr=8154#step:5:9

Not sure what `hasThemeJson` was meant to be, but it's not there.

**Comment by madhusudhand** [comment]
> Not sure what hasThemeJson was meant to be, but it's not there.

Issue fixed in #8174. I did a rebase, and it is a [success](https://github.com/Automattic/themes/actions/runs/10917831769/job/30301951487?pr=8154).

**Comment by iamtakashi** [comment]
Why does the theme URI need to be a dotcom site? Some themes could launch on dotorg first or not on dotcom at all.

**Pull request merged**
